### PR TITLE
changed template cache names to take into account the Twig C extension

### PR DESCRIPTION
**Pull request by fabpot**

### CODE REVIEW

**Comment by stof** [comment]
:+1:

File: CHANGELOG [hunk 1]
```diff
@@ -1,5 +1,6 @@
 * 1.22.2 (2015-XX-XX)
 
+ * changed template cache names to take into account the Twig C extension
```
Comment (Tobion) [inline CHANGELOG]: should be for 1.22.3 now

**Comment by Tobion** [comment]
:+1:  apart from the comment. I'd like to propose another PR on top of this when merged.

**Comment by stof** [comment]
@fabpot this broke the tests when the C extension is enabled

**Comment by ninsuo** [comment]
I don't understand why revoking the cache if the C extension is enabled.
- there are no difference in the generated template?
- shouldn't we take into account the `disable_c_ext`?

**Comment by Tobion** [comment]
The tests are going to be fixed by #1852

**Pull request merged**